test/lib/merkle_patricia_proof.js

```diff
@@ -37,7 +41,33 @@ contract('MerklePatriciaProof.verify()', (accounts) => {
         proofJson.account.rlpParentNodes,
         proofJson.account.stateRoot,
       );
-      assert.equal(proofStatus, true);
+      assert.strictEqual(proofStatus, true);
+
+      // Iterate over the stored data set and test all proofs in parallel.
+      // The data set was pre-generated with the fuzzy proof generator (see `./tools`).
+      await Promise.all(
+        generatedProofData.map(
```

Comment:
Just a suggestion, should this loop be moved above `it` statement and `it` statement description can also contain a pattern of proof like `ebl`. This way this will be evident what pattern has caused proof failure. Otherwise, it would be hard to track the pattern.

Reply:
Not sure what you mean. The "loop" is verifying each pattern with the library. How can I move that outside the `it` statement? 🤔

The pattern is actually not available here. Should we add the pattern to the proof data? It's a *little* extra effort. I don't think it's *too* much. Would also require changes in #743.

What do you think?

Reply:
I was something like this. 

```

  describe('Test Cases for Account proof verification', async () => {


    before(async () => {
      merklePatriciaLib = await MerklePatriciaProof.deployed();
    });


    it('Should pass when all the parameters are valid', async () => {
      const proofStatus = await merklePatriciaLib.verify.call(
        proofJson.account.sha3EncodedAccount,
        proofJson.account.hashedPath,
        proofJson.account.rlpParentNodes,
        proofJson.account.stateRoot,
      );
      assert.strictEqual(proofStatus, true);
      });


      generatedProofData.map(
        async (data) => {
       it(`Should pass when all the parameters are valid for the pattern ${data.pattern}` , async() => {
          const result = await merklePatriciaLib.verify.call(
            data.value,
            data.encodedPath,
            data.rlpParentNodes,
            data.root,
          );

          assert.strictEqual(
            result,
            true,
            `Could not verify data point: ${JSON.stringify(data)}`,
          );
        })
      })
  });
```

Advantage of this is, we will know for what kind of pattern the verification has failed. Otherwise, it would require manual analysis for this. 
Will it be easy to add pattern in the generate proof json?

Reply:
Like I said, the pattern is unfortunately no longer available at this point.

However, `${JSON.stringify(data)}` is part of the assert message and will print exactly the data that was responsible for the failing test. Is that what you mean?

Reply:
I understand data will be part of assert message. is there any easy way to identify what pattern was used to generate this data? 
I am emphasizing on the pattern because they are very useful to reproduce the scenario. Otherwise, someone needs to manually decode `data` to know the pattern.   

Do you think the pattern is a useful data point in debugging?

Reply:
I am proposing to add pattern in the proof data 👼

Reply:
#743 is already closed, so it should be a follow-up ticket ;)

Reply:
I will create one

Reply:
Created #754